backend/handler/db.js

```diff
@@ -100,41 +142,20 @@
 		return findResult.value;
 	};
 
-	function getURL(database) {
-		// Generate connection string that could include multiple hosts that
-		// represent a replica set.
-		let connectString = "mongodb://" + config.db.username + ":" + config.db.password + "@";
-		const hostPorts = [];
-
-		for(const host in config.db.host) {
-			hostPorts.push(config.db.host[host] + ":" + config.db.port[host]);
-		}
-
-		connectString += hostPorts.join(",");
-		connectString += "/" + database + "?authSource=admin";
-		connectString += config.db.replicaSet ? "&replicaSet=" + config.db.replicaSet : "";
-
-		if(Number.isInteger(config.db.timeout)) {
-			connectString += "&socketTimeoutMS=" + config.db.timeout;
-		}
-
-		return connectString;
-	}
-
 	Handler.getDB = function (database) {
-		if(db) {
+		if (db) {
 			return Promise.resolve(db.db(database));
 		} else {
 			return MongoClient.connect(getURL(database), connConfig).then(_db => {
 				db = _db;
-				return db;
+				return db.db(database);
 			});
 		}
 	};
 
 	Handler.getAuthDB = function () {
 		return MongoClient.connect(getURL("admin"), connConfig).then(_db => {
-			return _db;
+			return _db.db("admin");
```

Comment:
I don't know why we reconnect for authDB, it's a bit weird..

we should just return Handler.getDB("admin") ?